flume-ng-core/src/test/java/org/apache/flume/source/TestAvroSource.java

```diff
@@ -339,7 +339,34 @@ public void testSslRequest() throws InterruptedException, IOException {
     context.put("keystore", "src/test/resources/server.p12");
     context.put("keystore-password", "password");
     context.put("keystore-type", "PKCS12");
+
+    Configurables.configure(source, context);
+
+    doSslRequest();
+  }
+
+  @Test
+  public void testSslRequestWithGlobalKeystore() throws InterruptedException, IOException {
+
+    System.setProperty("javax.net.ssl.keyStore", "src/test/resources/server.p12");
+    System.setProperty("javax.net.ssl.keyStorePassword", "password");
+    System.setProperty("javax.net.ssl.keyStoreType", "PKCS12");
```

Comment:
That's interesting. Does everything support PKCS12 across the board? I guess it does, but we've never tested it.

Reply:
I've tested it all components works with PKCS12 (I guess because it is supported at JVM/JDK level).